kediatr-core/src/main/kotlin/com/trendyol/kediatr/AsyncPipelineBehavior.kt

```diff
@@ -0,0 +1,23 @@
+package com.trendyol.kediatr
+
+/**
+ * Interface to be implemented for a non-blocking pipeline behavior
+ *
+ * @since 1.0.12
+ * @see AsyncPipelineBehavior
+ */
+interface AsyncPipelineBehavior {
+    /**
+     * Process to invoke before handling any query, command or notification
+     *
+     * @param request the request to handle
+     */
+    suspend fun <TRequest>preProcess(request: TRequest)
```

Comment:
It seems that here we are defining an interface that gives an ability to return another request instance or return a modified instance. Should KediatR allow such behaviour since ? Or should we modify the following proccessing code in order to take account of such possible implementations.

    fun <TRequest, TResponse> processPipeline(pipelineBehaviors: Collection<PipelineBehavior>, request: TRequest, act: () -> TResponse): TResponse {
        pipelineBehaviors.forEach { it.preProcess(request) }
        val result = act()
        pipelineBehaviors.forEach { it.postProcess(request) }
        return result
    }